PMDMMO-common/src/com/darkxell/common/ai/visibility/Visibility.java

```diff
@@ -0,0 +1,112 @@
+package com.darkxell.common.ai.visibility;
+
+import java.util.HashSet;
+
+import com.darkxell.common.ai.AI;
+import com.darkxell.common.dungeon.data.FloorData;
+import com.darkxell.common.dungeon.floor.Tile;
+import com.darkxell.common.dungeon.floor.room.Room;
+import com.darkxell.common.item.ItemEffects;
+import com.darkxell.common.pokemon.DungeonPokemon;
+import com.darkxell.common.util.Direction;
+
+public class Visibility {
+
+    /** Reference to parent AI. */
+    public final AI ai;
+    /** The Tiles that this Pokemon can currently see. */
+    private HashSet<Tile> currentlyVisible = new HashSet<>();
+    /** The Tiles that were seen with an Item on it. */
+    private HashSet<Tile> itemTiles = new HashSet<>(); // TODO has to update when Items are moved
+    /** The Tiles that were already seen. */
+    private HashSet<Tile> seenTiles = new HashSet<>();
+
+    public Visibility(AI ai) {
+        this.ai = ai;
+    }
+
+    /** @return The Tiles this Pokemon can currently see from the Tile it's at. */
+    public HashSet<Tile> currentlyVisibleTiles() {
+        return new HashSet<>(this.currentlyVisible);
+    }
+
+    /** @return <code>true</code> if the input Tile has been seen in this Floor. */
+    public boolean hasSeenTile(Tile tile) {
+        return this.seenTiles.contains(tile);
+    }
+
+    /**
+     * @return <code>true</code> if the Item on the input Tile is visible by this Pokemon (doesn't check if there
+     * actually is an Item).
+     */
+    public boolean isItemVisible(Tile tile) {
```

Comment:
Would a better name be "canSeeItemTile" or "isItemTileVisible"? Something to emphasize that it is only checking whether it can, not that it is checking for an item.